### DISCUSSION TITLE
Dovecot ban everyone

### DISCUSSION THREAD

**Discussion by prince-jn** [fail2ban/fail2ban#3345] [Q&A] [answered]
Hello, I try to configure dovecot with fail2ban.

Dovecot version : 2.3.16 (7e2e900c1a)
Fail2ban version : Fail2Ban v0.11.2
Linux : rocky-release-8.6-3.el8.noarch

My config
```
[dovecot]
enabled = true
filter = dovecot
port = pop3,pop3s,imap,imaps
logpath = /var/log/maillog
```

Regex tried 
```
failregex = ^authentication failure; logname=<F-ALT_USER1>\S*</F-ALT_USER1> uid=\S* euid=\S* tty=dovecot ruser=<F-USER>\S*</F-USER> rhost=<HOST>(?:\s+user=<F-ALT_USER>\S*</F-ALT_USER>)?\s*$
            ^(?:Aborted login|Disconnected|Remote closed connection|Client has quit the connection)(?::(?: [^ \(]+)+)? \((?:auth failed, \d+ attempts(?: in \d+ secs)?|tried to use (?:disabled|disallowed) \S+ auth|proxy dest auth failed)\):(?: user=<<F-USER>[^>]*</F-USER>>,)?(?: method=\S+,)? rip=<HOST>(?:[^>]*(?:, session=<\S+>)?)\s*$
            ^pam\(\S+,<HOST>(?:,\S*)?\): pam_authenticate\(\) failed: (?:User not known to the underlying authentication module: \d+ Time\(s\)|Authentication failure \(password mismatch\?\)|Permission denied)\s*$
            ^[a-z\-]{3,15}\(\S*,<HOST>(?:,\S*)?\): (?:unknown user|invalid credentials|Password mismatch)
            <mdre-<mode>>
```
and 
```
failregex = ^authentication failure; logname=<F-ALT_USER1>\S*</F-ALT_USER1> uid=\S* euid=\S* tty=dovecot ruser=<F-USER>\S*</F-USER> rhost=<HOST>(?:\s+user=<F-ALT_USER>\S*</F-ALT_USER>)?\s*$
            ^(?:Aborted login|Disconnected|Remote closed connection|Client has quit the connection)(?:: (?:[^\(]+|\w+\([^\)]*\))+)? \((?:auth failed, \d+ attempts(?: in \d+ secs)?|tried to use (?:disabled|disallowed) \S+ auth|proxy dest auth failed)\):(?: user=<<F-USER>[^>]*</F-USER>>,)?(?: method=\S+,)? rip=<HOST>(?:[^>]*(?:, session=<\S+>)?)\s*$
            ^pam\(\S+,<HOST>(?:,\S*)?\): pam_authenticate\(\) failed: (?:User not known to the underlying authentication module: \d+ Time\(s\)|Authentication failure \([Pp]assword mismatch\?\)|Permission denied)\s*$
            ^[a-z\-]{3,15}\(\S*,<HOST>(?:,\S*)?\): (?:[Uu]nknown user|[Ii]nvalid credentials|[Pp]assword mismatch)
            <mdre-<mode>>
```

By default, dovecot log in /var/log/maillog. 
The problem is I get ban with this line
`Sep  4 09:27:06 jn dovecot[1695]: imap-login: Login: user=<x@y.net>, method=PLAIN, rip=173.x.x.x, lip=x.x.x.x, mpid=212946, TLS, session=<KfhH69nnUeWtsAt8>` which seems to be an info... not an error.

Is the regex bad or I need to change the log level / log info entries somewhere else? There's a lot of '?' which seems to say the error part is optional. 

Thank you!

**Comment by sebres** [comment, marked as answer]
> The problem is I get ban with this line

I don't see at all where dovecot filter can match this line.
Neither current [filter](https://github.com/fail2ban/fail2ban/blob/0.11/config/filter.d/dovecot.conf) (your second RE), nor original 0.11.2 [filter](https://github.com/fail2ban/fail2ban/blob/0.11.2/config/filter.d/dovecot.conf) (your first RE) would match this.

Even in aggressive mode it doesn't match:
```
$ fail2ban-regex 'Sep 4 09:27:06 jn dovecot[1695]: imap-login: Login: user=<x@y.net>, method=PLAIN, rip=192.0.2.111, lip=127.0.0.1, mpid=212946, TLS, session=<KfhH69nnUeWtsAt8>' 'dovecot[mode=aggressive]'

Running tests
=============

Use   failregex filter file : dovecot, basedir: /etc/fail2ban/
Use   filter options : {'mode': 'aggressive'}
Use      datepattern : {^LN-BEG}TAI64N
{^LN-BEG} : Default Detectors
Use      single line : Sep 4 09:27:06 jn dovecot[1695]: imap-login: Login...


Results
=======

Prefregex: 1 total
|  ^(?:\[\])?\s*(?:<[^.]+\.[^.]+>\s+)?(?:\S+\s+)?(?:kernel:\s?\[ *\d+\.\d+\]:?\s+)?(?:@vserver_\S+\s+)?(?:(?:(?:\[\d+\])?:\s+[\[\(]?(?:dovecot(?:-auth)?|auth)(?:\(\S+\))?[\]\)]?:?|[\[\(]?(?:dovecot(?:-auth)?|auth)(?:\(\S+\))?[\]\)]?:?(?:\[\d+\])?:?)\s+)?(?:\[ID \d+ \S+\]\s+)?(?:(?:dovecot: )?auth(?:-worker)?(?:\([^\)]+\))?: )?(?:pam_unix(?:\(dovecot:auth\))?: |(?:pop3|imap|managesieve|submission)-login: )?(?:Info: )?(?P<content>.+)$
`-

Failregex: 0 total

Ignoreregex: 0 total

Date template hits:
|- [# of hits] date format
|  [1] {^LN-BEG}(?:DAY )?MON Day %k:Minute:Second(?:\.Microseconds)?(?: ExYear)?
`-

Lines: 1 lines, 0 ignored, 0 matched, 1 missed
[processed in 0.00 sec]

|- Missed line(s):
|  Sep 4 09:27:06 jn dovecot[1695]: imap-login: Login: user=<x@y.net>, method=PLAIN, rip=192.0.2.111, lip=127.0.0.1, mpid=212946, TLS, session=<KfhH69nnUeWtsAt8>
`-
```

In my opinion this cannot be the line from your excerpt, so try to find what exactly causes the matches:
```bash
fail2ban-regex --print-all-matched /var/log/maillog dovecot
# or direct matched messages:
fail2ban-regex -o msg /var/log/maillog dovecot
```

**Reply by prince-jn** [reply]
Interesting. I wasn't aware of that cmd.
Like you said, there's no match with my file.
I guess fail2ban read an other file... which seems to be journalctl.
Is it default? Does it read my maillog + journalctl?
Thank you!

**Reply by prince-jn** [reply]
For information
Dovecot ["PAM"](https://doc.dovecot.org/configuration_manual/authentication/pam/) was activated. It was running before my db auth. So there was an error in the journalctl.

**Reply by sebres** [reply]
> I guess fail2ban read an other file... which seems to be journalctl.
> Is it default? Does it read my maillog + journalctl?

Then its `backend` is configured by default as `systemd` on that system.
More info in [man jail.conf.5](https://manpages.debian.org/testing/fail2ban/jail.conf.5.en.html) or [wiki :: Proper fail2ban configuration](https://github.com/fail2ban/fail2ban/wiki/Proper-fail2ban-configuration).
To check how your jail are configured you can dump current parameters with `fail2ban-client -d | grep "\['add'"`. Also `fail2ban-client status $jail` shows whether the jail monitoring log-file or journal.
And `fail2ban-regex` is also able to read from journal, just replace log-filename with `systemd-journal`.
